### ISSUE TITLE
Wrong GROUPING SETS implementation when using 2 empty grouping sets

### ISSUE THREAD

**Issue by lukaseder** [duckdb/duckdb#7219]
### What happens?

Duplicate grouping sets are not respected.

### To Reproduce

```sql
select count(*)
group by grouping sets ((), ())
```

This produces:

```
|count_star()|
|------------|
|1           |
```

But it should produce:

```
|count_star()|
|------------|
|1           |
|1           |
```

### OS:

Microsoft Windows [Version 10.0.22621.1555]

### DuckDB Version:

v0.7.2-dev2430

### DuckDB Client:

JDBC

### Full Name:

Lukas Eder

### Affiliation:

Data Geekery / jOOQ

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by lukaseder** [comment]
Note it seems that the problem is related to the empty grouping sets being consecutive without any other grouping sets being present. When adding at least one "normal" grouping set, then the problem disappears:

```sql
select count(*)
from (values (1)) t (a)
group by grouping sets ((), (), (a), (), ())
```

This results in:

```
|count_star()|
|------------|
|1           |
|1           |
|1           |
|1           |
|1           |
```

**Comment by Mytherin** [comment]
Thanks for the report! I have pushed a fix in #7234
